src/test/java/org/elasticsearch/index/query/BaseQueryTestCase.java

```diff
@@ -146,7 +152,8 @@ public void testFromXContent() throws IOException {
     @Test
     public void testToQuery() throws IOException {
         QueryParseContext context = new QueryParseContext(index, queryParserService);
-        assertLuceneQuery(this.testQuery, this.testQuery.toQuery(context));
+        context.setAllowUnmappedFields(true);
```

Comment:
Sorry for confusion, accidentaly deleted the previous comment. Short version was: I was trying to avoid complicated setup for the tests here. If setting mappings for specific fields is easy we should consider it here, otherwise the idea of the base test class so far was to mainly test parsing from Json and to check that serialization and query construction works. Checking that the constructed lucene queries really work on an index might be better in a single node / integration test or what do you think?